File: core/src/com/broll/gainea/server/core/cards/impl/play/C_MonsterSteer.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.Card;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.objects.Monster;
import com.broll.gainea.server.core.utils.SelectionUtils;
import com.broll.gainea.server.core.utils.UnitControl;
import com.google.common.collect.Lists;

import java.util.ArrayList;

public class C_MonsterSteer extends Card {
    public C_MonsterSteer() {
        super(72, "Monsterköder", "Wählt ein Monster und bewegt es um ein Feld weiter, mögliche Ziele werden dabei angegriffen.");
    }

    @Override
    public boolean isPlayable() {
        return game.getObjects().stream().anyMatch(it -> it instanceof Monster);
    }

    @Override
    protected void play() {
        Monster monster = SelectionUtils.selectWildMonster(game,"Wähle das Monster das bewegt werden soll");
        Location target = selectHandler.selectLocation("Wähle das Ziel (Einheiten werden angegriffen)", new ArrayList<>(monster.getLocation().getWalkableNeighbours()));
        UnitControl.conquer(game, Lists.newArrayList(monster), target);
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_MonsterSteer.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.cards.Card
import com.broll.gainea.server.core.objects.monster.Monster
import com.broll.gainea.server.core.utils.UnitControl.conquer
import com.broll.gainea.server.core.utils.selectWildMonster

class C_MonsterSteer : Card(72, "Monsterköder", "Wählt ein Monster und bewegt es um ein Feld weiter, mögliche Ziele werden dabei angegriffen.") {
    override val isPlayable: Boolean
        get() = game.objects.any { it is Monster }

    override fun play() {
        val monster: Monster = game.selectWildMonster("Wähle das Monster das bewegt werden soll")
                ?: return
        val target = selectHandler.selectLocation("Wähle das Ziel (Einheiten werden angegriffen)",
                monster.location.connectedLocations.toList())
        game.conquer(listOf(monster), target)
    }
}
